core/src/smartcontracts/isi/asset.rs

```diff
@@ -695,4 +695,25 @@ pub mod query {
             Ok(entry.owned_by() == &account_id)
         }
     }
+
+    impl ValidQuery for FindAssetRegisteringTransaction {
+        #[metrics(+"find_asset_registering_transaction")]
+        fn execute(&self, wsv: &WorldStateView) -> Result<Self::Output, Error> {
+            let asset_id = self
+                .id
+                .evaluate(wsv, &Context::default())
+                .wrap_err("Failed to get asset id")
+                .map_err(|e| Error::Evaluate(e.to_string()))?;
+            let account_id = self
+                .account_id
+                .evaluate(wsv, &Context::default())
+                .wrap_err("Failed to get account id")
+                .map_err(|e| Error::Evaluate(e.to_string()))?;
+            iroha_logger::trace!(%asset_id, %account_id);
+            let tx_hash = wsv
+                .map_account(&account_id, |account| account.asset_register_tx(&asset_id))?
+                .ok_or_else(|| FindError::Asset(asset_id))?;
+            Ok(tx_hash)
```

Comment:
I think query should respond with the actual hash of transaction, which registered asset. But, I believe, current implementation will return hash just before the one we need

Reply:
Why *believe* when you can test. @ilchu Consider adding a query test to `iroha_client`. Given that @astrokov7 is ignoring my requests for updates, we should make an exception.